ipynb/chromeos/ChromeOS_Tests.ipynb

```diff
@@ -0,0 +1,827 @@
+    "    :type get_token: bool\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    results_dir = \"~/results-dir\"\n",
+    "    pwr_file = \"~/power.txt\"\n",
+    "    \n",
+    "    test_cmd = 'test_that -b oak {} --results_dir {} {}\\n'.format(te.ip,\n",
+    "                                                                  results_dir,\n",
+    "                                                                  test)\n",
+    "    pwr_cmd = 'dut-control -t {} -y dvfs1_mw dvfs2_mv > {}\\n'.format(pwr_time_s,\n",
+    "                                                                     pwr_file)\n",
+    "    \n",
+    "    # Create cros_sdk session\n",
+    "    cros_sdk_session = CrosSdkSession(password)\n",
+    "        \n",
+    "    logging.info('#### Start %s execution', test)\n",
+    "    cros_sdk_session.stdin.write(test_cmd)\n",
+    "    \n",
+    "    if get_token:\n",
+    "        # Setup socket to get token from target\n",
+    "        sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)\n",
+    "        sock.bind((HOST_IP, 1234))\n",
+    "        logging.debug('     Waiting for token...')\n",
+    "        while True:\n",
+    "            data, addr = sock.recvfrom(32)\n",
+    "            if addr[0] == te.ip and data == \"POWER\":\n",
+    "                break\n",
+    "        sock.close()    \n",
+    "        logging.debug('     Token received....')\n",
+    "    \n",
+    "    logging.debug('     Start trace collection')\n",
+    "    # Check if trace events need to be collected\n",
+    "    if te.ftrace:\n",
+    "        te.ftrace.start()\n",
+    "        \n",
+    "    sleep(5)\n",
+    "    # Start measuring power\n",
+    "    cros_sdk_session.stdin.write(pwr_cmd)\n",
+    "    \n",
+    "    # communicate will close the session when the command terminates\n",
+    "    cros_sdk_session.communicate()\n",
+    "    logging.info('#### Completed %s execution', test)\n",
+    "    \n",
+    "    if te.ftrace:\n",
+    "        te.ftrace.stop()\n",
+    "        te.ftrace.get_trace(os.path.join(te.res_dir, 'trace.dat'))\n",
+    "    \n",
+    "    # Parse results using test-specific parser\n",
+    "    results = parse_results[test](results_dir)\n",
+    "    \n",
+    "    # Copy results to our Test Environment results directory\n",
+    "    with open(os.path.join(te.res_dir, 'results.json'), 'w') as outfile:\n",
+    "        json.dumps(results, outfile)\n",
+    "    \n",
+    "    return {\n",
+    "        \"results\"  : results,\n",
+    "        \"pwr_file\" : pwr_file\n",
+    "    }"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "collapsed": false,
+    "run_control": {
+     "marked": false
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "········\n"
+     ]
+    }
+   ],
+   "source": [
+    "# ask user for host password\n",
+    "password = getpass.getpass()"
```

Comment:
Does this requires to input the password each time?
Otherwise we can try to use the "standard" GNOME keyring to store and recover the password in an automated way each time we execute the cell.
Here is an example of code I use for my dotfiles:
   https://github.com/derkling/dotfiles/blob/master/bin/gnome_keyring.py
We just need the set_credentials and get_credentials functions.